q2_demux/tests/test_demux.py

```diff
@@ -1171,6 +1172,40 @@ def test_sequence_length_uses_subsample_paired(self):
                 self.assertEqual(payload["minSeqLen"]["forward"], 3)
                 self.assertEqual(payload["minSeqLen"]["reverse"], 5)
 
+    def test_empty_single_end(self):
+        empty = SingleLanePerSampleSingleEndFastqDirFmt(
```

Comment:
This creates an invalid format (see https://github.com/qiime2/qiime2/issues/490 for more details). Please fix the test data so that it passes the format's `.validate()` method. I suspect this is a problem for all the test data, so please check all test cases' backing data.